### PR TITLE
Drop support for older redis clients

### PR DESCRIPTION
**Pull request by agis**
Tested with [rafka end-to-end tests](https://github.com/skroutz/rafka/tree/master/test).

### CODE REVIEW

**Comment by agis** [comment]
We still have to decide a sensible default for reconnect attempts.

File: Gemfile.lock [hunk 1]
```diff
@@ -2,7 +2,7 @@ PATH
   remote: .
   specs:
     rafka (0.3.0)
-      redis (~> 3)
+      redis (~> 3, >= 3.3.2)
```
Comment (avgerin0s) [inline Gemfile.lock:5]: 3.3.5?

Comment (avgerin0s) [reply Gemfile.lock:5]: or change the commit message

File: lib/rafka.rb [hunk 2]
```diff
@@ -22,28 +22,4 @@ def self.wrap_errors
...
-  def self.with_retry(times: REDIS_DEFAULTS[:reconnect_attempts], every_sec: 1)
-    attempts = 0
-
-    begin
-      yield
-    rescue Redis::CannotConnectError => e
-      if Gem::Version.new(Redis::VERSION) < Gem::Version.new("3.2.2")
-        if attempts < times
-          attempts += 1
-          sleep every_sec
```
Comment (avgerin0s) [inline lib/rafka.rb:41]: The new implementation does not sleep for exactly `1s`. It sleeps for way less and increases it at each attempt.

Maybe you want to adjust the defaults

Comment (agis) [reply lib/rafka.rb:41]: Maybe you are talking about redis-rb 4.x? We depend on 3.3.x here.

File: Gemfile.lock (hunk 1 above)
Comment (agis) [reply Gemfile.lock:5]: Updated the commit message.

File: lib/rafka.rb (hunk 2 above)
Comment (agis) [reply lib/rafka.rb:41]: Nvm my previous comment. I've left the retries to 1 (no retries). Clients will have to set this explicitly from now on.

**Pull request merged**